workers/unity/Packages/com.improbable.gdk.tools/Common.cs

```diff
@@ -111,14 +113,29 @@ private static string DiscoverLocation(string binarybaseName)
 
             var splitPath = pathValue.Split(Path.PathSeparator);
 
-            if (Application.platform == RuntimePlatform.OSXEditor && !splitPath.Contains(UsrLocalBinDir))
+            if (Application.platform == RuntimePlatform.OSXEditor)
             {
-                splitPath = splitPath.Union(new[] { UsrLocalBinDir }).ToArray();
+                if (!splitPath.Contains(UsrLocalBinDir))
+                {
+                    var usrLocalBinPaths = new[] { UsrLocalBinDir, Path.Combine(UsrLocalBinDir, binarybaseName) };
+                    splitPath = splitPath.Union(usrLocalBinPaths).ToArray();
+                }
+
+                if (!splitPath.Contains(UsrLocalShareDir))
+                {
+                    var usrLocalSharePaths = new[] { UsrLocalShareDir, Path.Combine(UsrLocalShareDir, binarybaseName) };
+                    splitPath = splitPath.Union(usrLocalSharePaths).ToArray();
+                }
+            }
+
+            var location = splitPath.Select(p => Path.Combine(p, binarybaseName)).FirstOrDefault(File.Exists);
+            if (location != null)
+            {
+                return location;
             }
 
-            return splitPath
-                .Select(p => Path.Combine(p, binarybaseName))
-                .FirstOrDefault(File.Exists) ?? string.Empty;
+            Debug.LogError($"Could not discover location for {binarybaseName}");
```

Comment:
Not for this PR, but we need to come up with a prefix for SpatialOS prints so people have a hint of what's giving the errors.